src/libratbag-data.c

```diff
@@ -207,6 +215,69 @@ init_data_hidpp20(struct ratbag *ratbag,
 	}
 }
 
+static void
+init_data_sinowealth(struct ratbag *ratbag,
+		     GKeyFile *keyfile,
+		     struct ratbag_device_data *data)
+{
+	const char *devices_group = "Driver/sinowealth/devices/";
+	const char *group = "Driver/sinowealth";
+
+	GError *error = NULL;
+
+	size_t group_count = 0;
+	_cleanup_(g_strfreevp) char **groups = g_key_file_get_groups(keyfile, &group_count);
+
+	int disable_fallback = g_key_file_get_integer(keyfile, group, "DisableFallback", &error);
+	if (!disable_fallback) {
+		data->sinowealth.fallback_device = zalloc(sizeof(struct sinowealth_device_data));
+		struct sinowealth_device_data *device = data->sinowealth.fallback_device;
+
+		device->button_count = 6;
+		device->device_name = "Generic SinoWealth Mouse";
+		device->fw_version = "0000";
+		device->led_type = SINOWEALTH_LED_TYPE_RBG;
```

Comment:
feel free to ignore this: do you want to specify the fallback device in the respective device data file too? Because there's a chance they may need to differ anyway (eventually) between the two USB PIDs

Reply:
I don't think it's needed with the fact some of the mice with the same PIDs have different parameters, so we would have to do a lucky guess in every single device file.